### DISCUSSION TITLE
Take image of my renderer with react-three/fiber

### DISCUSSION THREAD

**Discussion by webgldevfun** [community/community#20996] [Discussions] [answered]
I'm working on a react-three/fiber project.
How to take a screenshot of my 3D scene and save it as an image file?

**Comment by venus01015** [comment, marked as answer]
I had a similar case with you.

You can check this.
https://discourse.threejs.org/t/take-image-of-my-render-with-js-and-save-it/17390

I solved it like this:
*********************************
export default function Scene() {
  
  const { progress } = useProgress()
  const screenshot = useStore( (s) => s.screenshot )
  const strDownloadMime = "image/octet-stream";
  const gl = useThree((state) => state.gl)

  useEffect(()=>{
    useStore.setState({ loading: progress });
  },[progress])
  
  function saveFile (strData: string, filename: string) {
    var link = document.createElement('a');
    if (typeof link.download === 'string') {
        document.body.appendChild(link); //Firefox requires the link to be in the body
        link.download = filename;
        link.href = strData;
        link.click();
        document.body.removeChild(link); //remove the link when done
    } else {
        // location.replace(uri);
    }
  }

  useEffect(()=>{
    if(screenshot)
    {
      var imgData, imgNode;

      try {
          var strMime = "image/jpeg";
          imgData = gl.domElement.toDataURL(strMime);

          saveFile(imgData.replace(strMime, strDownloadMime), "test.jpg");

      } catch (e) {
          console.log(e);
          return;
      }

      useStore.setState({screenshot:false})
    }
  }, [screenshot])  

  return (
    <React.Fragment>

        <color attach="background" args={['#dddddd']}/>
        {/* <axesHelper args={[3]}/> */}
        <Environment preset='apartment'/>

        <Suspense fallback={null}>
          <Scene1/>
          <Avatar1/>
        </Suspense>
    </React.Fragment>   
  )
}
*********************************

**Reply by webgldevfun** [reply]
Thank you for your answer
